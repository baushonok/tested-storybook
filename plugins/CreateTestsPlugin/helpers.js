const fs = require("fs");
const path = require("path");

const getComponentName = (fileContent, pattern) =>
  fileContent.match(pattern)[0];

const getComponentStoriesNames = (fileContent, pattern) =>
  fileContent.match(pattern);

const generateHermioneTest = (
  testDirectoryPath,
  componentName,
  componentStoryName,
  testTemplate
) => {
  const testPath = path.resolve(
    testDirectoryPath,
    `${componentStoryName}.hermione.js`
  );

  if (!fs.existsSync(testPath)) {
    fs.createWriteStream(testPath, "utf8");
    fs.writeFileSync(
      testPath,
      testTemplate(componentName, componentStoryName),
      "utf8"
    );
  }
};

module.exports = {
  getComponentName,
  getComponentStoriesNames,
  generateHermioneTest
};
