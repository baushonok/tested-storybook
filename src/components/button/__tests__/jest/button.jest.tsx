import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';

import { getButton } from './helpers';
import Button from '../..';

describe('Button', () => {
  describe('shapshots', () => {
    test('should match when empty', () => {
      const tree = renderer.create(<Button />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('should match with text', () => {
      const tree = renderer.create(<Button>Click me!</Button>).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('should match with className', () => {
      const tree = renderer
        .create(<Button className="jest">Click me!</Button>)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('should match when disabled', () => {
      const tree = renderer
        .create(<Button disabled>Click me!</Button>)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('behaviour', () => {
    const buttonText = 'Click me!';
    const clickHandler = jest.fn();

    test('should invoke onClick', () => {
      render(<Button onClick={clickHandler}>{buttonText}</Button>);
      const button = getButton(screen);

      button.click();

      expect(clickHandler).toHaveBeenCalledTimes(1);
    });

    test('should not invoke onClick when disabled', () => {
      render(
        <Button disabled onClick={clickHandler}>
          {buttonText}
        </Button>
      );
      const button = getButton(screen);

      button.click();

      expect(clickHandler).not.toHaveBeenCalled();
    });

    test('should not invoke onClick when not function', () => {
      const fakeClickHandler = undefined;
      render(<Button onClick={fakeClickHandler}>{buttonText}</Button>);
      const button = screen.getByText(buttonText);

      button.click();

      expect(clickHandler).not.toHaveBeenCalled();
    });
  });
});
