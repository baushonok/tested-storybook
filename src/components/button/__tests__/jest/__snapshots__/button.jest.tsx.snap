// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button shapshots should match when disabled 1`] = `
<button
  className="button "
  disabled={true}
  onClick={[Function]}
  type="button"
>
  Click me!
</button>
`;

exports[`Button shapshots should match when empty 1`] = `
<button
  className="button "
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`Button shapshots should match with className 1`] = `
<button
  className="button jest"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Click me!
</button>
`;

exports[`Button shapshots should match with text 1`] = `
<button
  className="button "
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Click me!
</button>
`;
